### PR TITLE
JWT: Add support for assigning org roles

### LINKED ISSUE #51783: JWT: Add support for assigning org roles
Implement org role mapping for JWT Authentication.

Approach:

- `role_attribute_path` config setting similar to the one present in oauth (generic, github, gitlab)

### PR DESCRIPTION
**Pull request by nrwiersma**
**What this PR does / why we need it**:

Implement org role mapping for JWT Authentication similar to the one present in oAuth. This includes the `role_attribute_path`, `role_attribute_strict` as well as the planned `allow_assign_grafana_admin` as discussed in the issue.

**Which issue(s) this PR fixes**:

Fixes #51783

**Special notes for your reviewer**:

I have assumed the property for the new Grafana Admin assignment will be named `allow_assign_grafana_admin`.

### CODE REVIEW

File: conf/defaults.ini [hunk 1]
```diff
@@ -611,7 +611,10 @@ jwk_set_file =
 cache_ttl = 60m
 expected_claims = {}
 key_file =
-auto_sign_up = false
+role_attribute_path =
```
Comment (Jguer) [inline conf/defaults.ini:614]: :+1: I think you can also add these to the `sample.ini`

File: docs/sources/setup-grafana/configure-security/configure-authentication/jwt.md [hunk 2]
```diff
@@ -99,3 +99,68 @@ You might also want to validate that other claims are really what you expect the
...
+}
+```
+
+Config:
+
+```bash
+role_attribute_path = contains(info.roles[*], 'admin') && 'Admin' || contains(info.roles[*], 'editor') && 'Editor' || 'Viewer'
+```
+
+### Grafana Admin Role
```
Comment (Jguer) [inline docs/sources/setup-grafana/configure-security/configure-authentication/jwt.md:164]: nice docs

**Comment by Jguer** [comment]
I think once the issues are fixed with the ini files (there's also a small conflict with main after the sample.ini change) it's good to merge. 

I'll take some time tomorrow to play around with it before approving but I think code wise it's good

**Comment by nrwiersma** [comment]
I will rebase in the morning (GMT+2) unless it will help you for me to do it now?

**Comment by Jguer** [comment]
While trying locally I found some issues with creating users to begin with. For existing users it worked well :+1: 

I've added 2 commits:

- Fix new user creation with roles 
- Add this option to the existing jwt_proxy devenv

If you have no issues with these @nrwiersma , I'll merge this PR in around 20h

File: devenv/docker/blocks/auth/jwt_proxy/docker-compose.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@
       POSTGRES_USER: keycloak
       POSTGRES_PASSWORD: password
     volumes:
-      - ./docker/blocks/jwt_proxy/cloak.sql:/docker-entrypoint-initdb.d/cloak.sql
+      - ./docker/blocks/auth/jwt_proxy/cloak.sql:/docker-entrypoint-initdb.d/cloak.sql
```
Comment (Jguer) [inline devenv/docker/blocks/auth/jwt_proxy/docker-compose.yaml:9]: @linoman had to fix this here. There's always one that slips through the cracks :smile:

**Review by Jguer — APPROVED** [review]

File: devenv/docker/blocks/auth/jwt_proxy/docker-compose.yaml (hunk 3 above)
Comment (linoman) [reply devenv/docker/blocks/auth/jwt_proxy/docker-compose.yaml:9]: I believe I forgot about the volumes. I just created a PR to address this.

https://github.com/grafana/grafana/pull/54828

**Comment by Jguer** [comment]
Merged @nrwiersma , releasing with Grafana 9.2 :slightly_smiling_face: , thanks for your contribution.

**Pull request merged**